### PR TITLE
Align run name strategy between MLflowCallback and track_in_mlflow method

### PR DESCRIPTION
**Pull request by TTRh**
## Motivation
The strategy for infering the run name was not aligned between the base class `MLflowCallback` and the decorator `track_in_mlflow`. 

* in MLflowCallback we take the run name given by the user and we fallback to trial number if not given : https://github.com/optuna/optuna-integration/blob/main/optuna_integration/mlflow.py#L149 
* in decorator we take by default trial.number: https://github.com/optuna/optuna-integration/blob/main/optuna_integration/mlflow.py#L211 

## Description of the changes

This PR aligned the strategy to get the run name from MLfowCallback and pass it to mlflow. We lookup `_mlflow_kwargs` and we fallback on trial.number if not given.

### CODE REVIEW

**Comment by TTRh** [comment]
Tests are failing but i'm not sure it's related to my change i see in the logs: 
```
ERROR collecting tests/lightgbm_tuner_tests/test_alias.py
```

**Comment by HideakiImamura** [comment]
Sorry for the confusion. I will fix the main branch. Please wait a minuite.

**Comment by HideakiImamura** [comment]
The main branch has been fixed in https://github.com/optuna/optuna-integration/pull/112. Could you merge the main branch?

File: optuna_integration/mlflow.py [hunk 1]
```diff
@@ -207,8 +207,9 @@ def wrapper(trial: optuna.trial.Trial) -> float | Sequence[float]:
                     study = trial.study
                     self._initialize_experiment(study)
                     nested = self._mlflow_kwargs.get("nested")
+                    run_name = self._mlflow_kwargs.get("run_name") or str(trial.number)
```
Comment (nzw0301) [inline optuna_integration/mlflow.py:210]: If the first value can be evaluated as `False` although I think this happens very unlikely, `str(trial.number)` is used, so how about using `str(trial.number)` as a default?

```suggestion
                    run_name = self._mlflow_kwargs.get("run_name", str(trial.number))
```

Comment (TTRh) [reply optuna_integration/mlflow.py:210]: sure, let's do that

Comment (nzw0301) [reply optuna_integration/mlflow.py]: Ah, right. Thank you for your comment. Maybe we can update [`MLFlowCallback`](https://github.com/optuna/optuna-integration/blob/1c3fa01823c4e448d3773c63e7fd27e45e016b5c/optuna_integration/mlflow.py#L149)'s part too.

Comment (nzw0301) [reply optuna_integration/mlflow.py]: Thanks. Sorry, the mlflow test failed due to my suggestion, let me check what was wrong...

Comment (TTRh) [reply optuna_integration/mlflow.py]: yep i updated the test because we were setting run_name at None in the test

Comment (nzw0301) [reply optuna_integration/mlflow.py]: Awesome!

**Review by eukaryo — APPROVED** [review]
I have reviewed this PR. LGTM. Since the assigned reviewer is currently busy, I will proceed with merging.

**Pull request merged**